Spigot-Server-Patches/0551-Optimize-Villagers.patch

```diff
@@ -174,31 +194,43 @@ index a6d8ef5eb44f3f851a3a1be4032ca21ab1d7f2b2..c3e8a0145d63843736d2060f978cdf38
          }
      }
  
+diff --git a/src/main/java/net/minecraft/server/SectionPosition.java b/src/main/java/net/minecraft/server/SectionPosition.java
+index 7806f3c351cba3f0388da11888f900c48004dadf..cd0bda647780ce82ac94c8448543d5c040682397 100644
+--- a/src/main/java/net/minecraft/server/SectionPosition.java
++++ b/src/main/java/net/minecraft/server/SectionPosition.java
+@@ -145,6 +145,7 @@ public class SectionPosition extends BaseBlockPosition {
+         return (((long) i & 4194303L) << 42) | (((long) j & 1048575L)) | (((long) k & 4194303L) << 20); // Paper - Simplify to reduce instruction count
+     }
+ 
++    public long asLong() { return s(); } // Paper - OBFHELPER
+     public long s() {
+         return (((long) getX() & 4194303L) << 42) | (((long) getY() & 1048575L)) | (((long) getZ() & 4194303L) << 20); // Paper - Simplify to reduce instruction count
+     }
 diff --git a/src/main/java/net/minecraft/server/VillagePlaceRecord.java b/src/main/java/net/minecraft/server/VillagePlaceRecord.java
-index 1e9d7a3f902eb4571b93bb0e58cba966365f07b8..44535a3e2c3320aac472c5a7ee557fac7bab2530 100644
+index e5d2304808b2bb93041523968f0b88e020c61c1e..549f3da8e3bd33e3dc95486a40a64c0694306e48 100644
 --- a/src/main/java/net/minecraft/server/VillagePlaceRecord.java
 +++ b/src/main/java/net/minecraft/server/VillagePlaceRecord.java
-@@ -32,6 +32,7 @@ public class VillagePlaceRecord implements MinecraftSerializable {
-         return dynamicops.createMap(ImmutableMap.of(dynamicops.createString("pos"), this.a.a(dynamicops), dynamicops.createString("type"), dynamicops.createString(IRegistry.POINT_OF_INTEREST_TYPE.getKey(this.b).toString()), dynamicops.createString("free_tickets"), dynamicops.createInt(this.c)));
+@@ -34,6 +34,7 @@ public class VillagePlaceRecord {
+         this(blockposition, villageplacetype, villageplacetype.b(), runnable);
      }
  
-+    protected final boolean decreaseVacancy() { return b(); } // Paper - OBFHELPER
++    public final boolean decreaseVacany() { return b(); } // Paper - OBFHELPER
      protected boolean b() {
          if (this.c <= 0) {
              return false;
-@@ -42,6 +43,7 @@ public class VillagePlaceRecord implements MinecraftSerializable {
+@@ -44,6 +45,7 @@ public class VillagePlaceRecord {
          }
      }
  
-+    protected final boolean increaseVacancy() { return c(); } // Paper - OBFHELPER
++    public final boolean increaseVacany() { return c(); } // Paper - OBFHELPER
      protected boolean c() {
          if (this.c >= this.b.b()) {
              return false;
-@@ -52,14 +54,17 @@ public class VillagePlaceRecord implements MinecraftSerializable {
+@@ -54,14 +56,17 @@ public class VillagePlaceRecord {
          }
      }
  
-+    public final boolean hasVacancy() { return d(); } // Paper - OBFHELPER
++    public final boolean hasVacany() { return d(); } // Paper - OBFHELPER
```

Comment:
_ah yes, the vacany_

Reply:
Whoops, typos 
Had to reimplement the whole patch by hand :eyes: